pkg/controller/factory/builder.go

```diff
@@ -66,11 +66,11 @@ func BuildRSM(cluster *appsv1alpha1.Cluster, synthesizedComp *component.Synthesi
 		labels = constant.GetKBWellKnownLabels(clusterDefName, clusterName, compName)
 		compDefLabel = constant.GetClusterCompDefLabel(clusterCompDefName)
 	}
+	mergeLabels := intctrlutil.MergeMetadataMaps(labels, compDefLabel, synthesizedComp.Labels)
```

Comment:
passing custom labels in this way may have additional side effects, it involves updating to the pod template, whereas previously, the pods were updated directly

Reply:
> passing custom labels in this way may have additional side effects, it involves updating to the pod template, whereas previously, the pods were updated directly

revert it